src/core/Params.js

```diff
@@ -12,17 +14,21 @@ export default class Params {
 
     /**
      * Returns param value that matches given placeholder with param key.
-     * @param {String} placeholder
+     * @param {Object} token
+     *   @param {String} type Option from tokenTypes
+     *   @param {String} value Placeholder value
      * @return {String} param
      */
-    get(placeholder) {
+    get({type, value}) {
         if (!this.params) {
-            return placeholder;
+            return value;
         }
-        const matches = placeholder.match(/\w+(\s+\w+)*/);
+        const key = type === tokenTypes.PLACEHOLDER_STRING ?
+            value.substring(2).slice(0, -1) :
+            value.substring(1);
 
-        if (matches) {
-            return this.params[matches[0]];
+        if (key) {
+            return this.params[key];
```

Comment:
We really have three placeholder token types: STRING, INDENT, INDEXED. Currently the INDEXED and INDENT are thrown together (although they're really quite different). By having three distinct types, we could express the three different choices in much clearer way:

```js
switch(type) {
    case INDEXED:
    case STRING:
    case IDENT:
}
```

Also, I think parsing the contents of the string and indent placeholders should really belong to Tokenizer. The Formatter should not care about the specific syntax that placeholders have. It's also common practice for Lexer to provide both a raw value and some computed value of a token like string. In that scenario we actually won't need 3 token types as STRING and INDENT can be represented with one.

Finally I still think that we should take at least some minimal care about escaping. As the Lexer allows escape sequences like `\"` in strings, the parameter replacement should also support them.